### DISCUSSION TITLE
How to add service-specific health checks?

### DISCUSSION THREAD

**Discussion by Kralizek** [microsoft/aspire#1292] [Q&A] [answered]
Currently health checks are registered in the `AddServiceDefaults` method that invokes a `AddDefaultHealthChecks`.

I now want to add service-specific health checks. Some examples:
- backend => if database is available
- frontend => if backend is reachable

What's the suggested way to do this?

### Option 1
- edit `AddServiceDefaults` so that it doesn't invoke `AddDefaultHealthChecks`
- change `AddDefaultHealthChecks` so that it returns `IHealthChecksBuilder` instead of `IHostApplicationBuilder`
- in `Program.cs`, have something like

```csharp
builder.AddDefaultHealthChecks()
    .AddUrlGroup(new Uri("http://api/_health"), "api");
```

### Option 2
- in `Program.cs` call again the `AddHealthChecks` method and add own check

```csharp
builder.Services.AddHealthChecks()
    .AddUrlGroup(new Uri("http://api/_health"), "api");
```

Considering `AddHealthChecks` is already called via `AddServiceDefaults`, does it create a conflict?

**Comment by DamianEdwards** [comment, marked as answer]
Option 2 is the right approach. The [eShopLite sample in the aspire-samples repo has some extra custom extension methods that demonstrate this](https://github.com/dotnet/aspire-samples/blob/main/samples/eShopLite/eShopLite.ServiceDefaults/ServiceReferenceExtensions.cs#L69).

**Reply by Kralizek** [reply]
Slightly related. Is there a plan to have anything like a Aspire Component that registers health checks for project dependencies? In the link you provided there is a hand-crafted method.

**Reply by DamianEdwards** [reply]
See #898
